editor/create_dialog.cpp

```diff
@@ -183,6 +183,10 @@ void CreateDialog::_update_search() {
 			candidates.push_back(I->get());
 		}
 	}
+	// Also check the base type.
+	if (empty_search || search_text.is_subsequence_ofn(base_type)) {
+		candidates.push_back(base_type);
+	}
```

Comment:
I believe that instead of this additional check, it would be better to remove in `CreateDialog::_should_hide_type` the check, that excludes the base type. As done in #58744.

Reason is, that the base_type doesn't need to be treated in this case in a special way for determining, if it should be added to the `candidates` list. So the same effect can be achieved with less code.